src/test/java/gradle/cucumber/Request.java

```diff
@@ -4,14 +4,18 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.PrintWriter;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
 import java.net.Socket;
 
 public class Request {
 
   private final Socket socket;
 
-  public Request(String address, String port) throws IOException {
-    this.socket = new Socket(address, Integer.parseInt(port));
+  public Request(String port) throws IOException {
```

Comment:
Have you looked into using a proper HTTP client for this?

Reply:
I will look into that now!